airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json

```diff
@@ -64,11 +56,68 @@
         "title": "Username",
         "order": 5
       },
-      "password": {
-        "description": "The password associated with the username.",
-        "type": "string",
-        "airbyte_secret": true,
-        "title": "Password",
```

Comment:
Does removing this field make the new connector version non-backward compatible with older configurations without migrating the old configs? (or update them?)

Can we tolerate config with the old password field being at this location in the spec? (making sure with a test too?)

Reply:
Hi @ChristopheDuong. 
Nope, we have AdditionalProperties is set to true. 
https://github.com/airbytehq/airbyte/pull/11093/files#diff-e1acb96aaf759d23333bec115cba9c1d1af77282bdd7dcdc1346240f7f871d01L3858

So it's not explicitly described in spec, but we handle this in java code:
https://github.com/airbytehq/airbyte/pull/11093/files#diff-0fd1ff4a11af6edb89860081578c13598aa6cdb289c3e6ceae2a9d44eb708b80R95
`    if (credentials != null && credentials.has("auth_type") && "OAuth2.0".equals(
        credentials.get("auth_type").asText())) {
      LOGGER.debug("OAuth login mode is used");
...
...
...

    } else if (credentials != null && credentials.has("password")) {
      LOGGER.debug("User/password login mode is used");
      // Username and pass login option is selected on UI
      dataSource.setUsername(username);
      dataSource.setPassword(credentials.get("password").asText());

    } else {
      LOGGER.warn(
          "Obsolete User/password login mode is used. Please re-create a connection to use the latest connector's version");
      // case to keep the backward compatibility
      dataSource.setUsername(username);
      dataSource.setPassword(config.get("password").asText());
    }`


and in normalization side:
https://github.com/airbytehq/airbyte/pull/11093/files#diff-76b45d3c5541ea23ab17d448c349c37f1a4d12bf1a604cfa7cceedb130588facR236
`  if credentials.get("auth_type") == "OAuth2.0":
            dbt_config["authenticator"] = "oauth"
            dbt_config["oauth_client_id"] = credentials["client_id"]
            dbt_config["oauth_client_secret"] = credentials["client_secret"]
            dbt_config["token"] = credentials["refresh_token"]
        elif credentials.get("password"):
            dbt_config["password"] = credentials["password"]
        else:
            dbt_config["password"] = config["password"]
        return dbt_config`


I also added a test to check it:
https://github.com/airbytehq/airbyte/pull/11093/files#diff-aa693e6356d71e05c008807f89cc10f5f2bfb495c14086472892cd08ca02436eR175

Reply:
Ok, this sounds good!

I just arrived on this PR because I couldn't figure out why the snowflake tests were failing on other branches... it might have been nice to keep the old password field in the secret of the CI, so other branches without your changes wouldn't start failing though